### ISSUE TITLE
msgpack-lite decode failed with Invalid type

### ISSUE THREAD

**Issue by muz3** [mcollina/aedes-persistence-level#19]
Hi team, msgpack encode() and decode() work with Buffer type, but when retained loaded from my local levelDB, it gives ArrayBuffer, which caused decode() to fail and crash my mqtt broker. So to play safe, we need wrap the input of decode with either Uint8Array or Buffer. 

You can find out the same issue reported to msgpack-lite here. (https://github.com/kawanet/msgpack-lite/issues/44). I am currently downgrade my version from v6 to v5.1. Please let me know if you are okay with me to submit a MR. 

thanks,

Rui

=========Error Log=====
Error: Invalid type: 0x?
    at Codec.decode (../node_modules/msgpack-lite/lib/read-core.js:24:22)
    at DecodeBuffer.fetch (../node_modules/msgpack-lite/lib/decode-buffer.js:26:21)
    at DecodeBuffer.read (../node_modules/msgpack-lite/lib/flex-buffer.js:166:28)
    at Object.decode (../node_modules/msgpack-lite/lib/decode.js:10:18)
    at DestroyableTransform._transform (../node_modules/aedes-persistence-level/persistence.js:77:26)

**Comment by mcollina** [comment]
Can you please add a unit test/example to reproduce?

**Comment by muz3** [comment]
Sure I will find some time today or tomorrow.

**Comment by muz3** [comment]
Hi Matt,

I tried to use storeRetained ad createRetainedStream methods in unit test, but I was not able to reproduce the error. So I created an example here (https://github.com/RuiWangGit/msgpack-decode-bug). 

1. clone my repo , npm install and navigate to the folder
2. 'node broker.js ' to start the server
3.  'node subscriber.js' to start the subscriber, it will publish a random message upon subscribing.
4. disconnect the subscriber, then restart it. The retained message will crash broker. (Error: Invalid type: 0x�)

let me know if you have any questions.
